lib/active_set/column_instruction.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require_relative './attribute_instruction'
+
+class ActiveSet
+  class ColumnInstruction
+    def initialize(instructions_hash, item)
+      @instructions_hash = instructions_hash.symbolize_keys
+      @item = item
+    end
+
+    def key
+      return @instructions_hash[:key] if @instructions_hash.key? :key
+
+      titleized = attribute_instruction.keypath.map(&:titleize).join(' ')
+      return titleized unless attribute_instruction.attribute
+
+      attribute_resource = attribute_instruction.resource_for(item: @item)
+      return titleized unless attribute_resource
+      return titleized unless attribute_resource.class.respond_to?(:human_attribute_name)
+
+      attribute_resource.class.human_attribute_name(attribute_instruction.attribute)
+    end
+
+    def value
```

Comment:
The value for a column instruction will be either an executed Proc (with the item provided to the Proc), a value returned by treating the `:value` entry as an attribute path, or some default value (`—` in this class).